### DISCUSSION TITLE
Dictionaries from public

### DISCUSSION THREAD

**Discussion by RobAndrewHurst** [GEOLYTIX/xyz#2152] [Ideas]
We need to dynamically import languages from the public directory, based on the params provided.

**Comment by dbauszus-glx** [comment]
Having various dictionary merges inside many modules inside the core library is hard to maintain and not efficient. The merge happens when the library is loaded, not when the library is bundled.

Adding additional language entries is difficult due to the multiple entry points. Duplicate entries are likely to happen.

Additional languages unnecesarily increase the bundle size.

Any solution should be backwards compatible.

A plugin merging custom dictionary entries into the dictionaries object should still be able to do so.

The english fallback dictionary should always be loaded and should be complete. Adding language entries for other languages than english only is not desired.

The idea is that only the English dictionary is bundled.

Other dictionary are placed in the public directory.

A utility method can load and merge a language dictionary from the public dir into the dictionaries object.

**Comment by dbauszus-glx** [comment]
This has been resolved with this PR. https://github.com/GEOLYTIX/xyz/pull/2213
